### PR TITLE
QVAC-17062 feat[api]: add streaming transcription to bci addon

### PR DESCRIPTION
**Pull request by ishanvohra2**
**Note**: be concise and prefer bullet points.

## 🎯 What problem does this PR solve?

- BCI transcription was batch-only — callers had to buffer the entire neural signal and wait for a full decode before seeing any output, which is unusable for near-real-time UIs.
- Closes the "Streaming transcription is not implemented" known limitation from `0.1.0` (QVAC-17062).

## 📝 How does it solve it?

- Adds `BCIWhispercpp#transcribeStream(neuralStream, streamOpts)` that accepts an async iterable / `Uint8Array` / chunk array and returns a standard `QvacResponse` with incremental `onUpdate` callbacks.
- Pure-JS sliding-window driver over the existing batch `runJob` entrypoint — no native addon or binding changes. Each window is decoded in sequence and merged with the prior window via a word-level longest-suffix/longest-prefix stitcher.
- Segment-aware stitcher preserves native per-segment timestamps (`t0`/`t1`) on `emit:'delta'` updates and anchors them via a new `windowStartTimestep` field.
- `cancel()` / `unload()` / `destroy()` are stream-aware: they reject any in-flight window decode, fail the active stream response, and await the driver unwind before tearing down the addon — no dangling promises on teardown.
- Four new typed error codes for stream-specific failure modes (`STREAM_ALREADY_ACTIVE`, `INVALID_STREAM_INPUT`, `INVALID_STREAM_HEADER`, `WINDOW_TOO_LARGE`).

## 🧪 How was it tested?

- 21 unit tests / 68 asserts in `test/unit/stitch.test.js` covering `stitchMerge` and the segment-aware `stitchSegments` (overlap detection, timestamp preservation, `maxWords` cap, punctuation/case normalization, empty-input edge cases).
- 3 new integration tests (gated on `BCI_REQUIRE_MODEL`) covering single-window streaming, multi-window sliding with `emit:'full'`, and first-update-before-stream-end incrementality.
- Manually verified end-to-end via `examples/transcribe-stream-neural.js` feeding a `.bin` fixture in 4 KB chunks.

## 🔌 API Changes

```js
const response = await bci.transcribeStream(neuralChunkStream, {
  windowTimesteps: 1500, // ~30s window
  hopTimesteps: 500,     // ~10s hop
  emit: 'delta'          // or 'full' for the full running transcript
})

response.onUpdate(segments => {
  for (const s of segments) {
    // native t0/t1 preserved; windowStartTimestep anchors to stream timeline
    console.log(s.windowStartTimestep, s.t0, s.t1, s.text)
  }
})

const finalOutput = await response.await()
```

### CODE REVIEW

File: packages/bci-whispercpp/index.js [hunk 1]
```diff
@@ -183,6 +243,300 @@ class BCIWhispercpp {
...
+      if (channels !== null) {
+        const bufferedTs = Math.floor(bodyBytes / bytesPerTimestep)
+        if (bufferedTs > lastDecodedEndTs && bufferedTs > windowStartTs) {
+          await decodeRange(windowStartTs, bufferedTs - windowStartTs)
+        }
+      }
+
+      if (!this._streamAborted) {
+        this._streamResponse = null
+        response.ended()
```
Comment (ogad-tether) [inline packages/bci-whispercpp/index.js]: `emit:'full'` looks incorrect here. Because each window already calls `response.updateOutput([{ text: merged }])`, ending without an explicit result means `response.await()` resolves to the entire history of intermediate full transcripts, not just the final transcript. That contradicts the new API docs/changelog and will duplicate text for callers that flatten the resolved value. Could we end with the last merged snapshot explicitly (or otherwise separate incremental updates from the terminal result)?

File: packages/bci-whispercpp/lib/stream.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  for (const { seg, words } of perSegment) {
+    if (words.length === 0) continue
+    if (toSkip >= words.length) { toSkip -= words.length; continue }
+    if (toSkip === 0) {
+      deltaSegments.push({ ...seg, windowStartTimestep })
+    } else {
+      const remainingWords = words.slice(toSkip)
+      deltaSegments.push({
+        ...seg,
+        text: remainingWords.join(' '),
```
Comment (ogad-tether) [inline packages/bci-whispercpp/lib/stream.js:174]: When overlap trimming happens inside a segment, this rewrites `text` but preserves the original `t0`/`t1`. That means the emitted timestamps now describe the dropped words rather than the surviving tail, so `windowStartTimestep + t0/t1` becomes misleading for consumers. I think we either need to recompute/approximate the trimmed segment timing or avoid preserving native timings for partially-trimmed segments.

File: packages/bci-whispercpp/index.js [hunk 3]
```diff
@@ -183,6 +243,300 @@ class BCIWhispercpp {
...
+        if (channels === null) {
+          if (headerCarry.byteLength > 0) {
+            const combined = new Uint8Array(headerCarry.byteLength + chunk.byteLength)
+            combined.set(headerCarry, 0)
+            combined.set(chunk, headerCarry.byteLength)
+            chunk = combined
+            headerCarry = new Uint8Array(0)
+          }
+          if (chunk.byteLength < 8) {
+            headerCarry = chunk
```
Comment (ogad-tether) [inline packages/bci-whispercpp/index.js:380]: This buffers a partial header, but if the stream ends before the remaining header bytes arrive we currently fall through to a successful `response.ended()` with `channels === null`. Since this PR introduces `INVALID_STREAM_HEADER`, shouldn't EOF with a non-empty `headerCarry` fail instead of looking like a clean empty stream?

**Review by ogad-tether — CHANGES_REQUESTED** [review]
I found three correctness issues that should be fixed before this lands:

- `emit:'full'` currently makes `response.await()` resolve to the full history of intermediate snapshots instead of the final transcript only.
- `stitchSegments()` rewrites partially-overlapped segment text without adjusting its timestamps, so emitted `t0`/`t1` no longer match the surviving words.
- A stream that ends before the full 8-byte header arrives is treated as a clean empty stream instead of surfacing `INVALID_STREAM_HEADER`.

The inline comments below point to the relevant spots.

File: packages/bci-whispercpp/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,40 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.1.1]
```
Comment (GustavoA1604) [inline packages/bci-whispercpp/CHANGELOG.md]: No need to bump version as we have still not published the original package. Just ensure that CHANGELOG.md is up to date in 0.1.0

**Review by GustavoA1604 — CHANGES_REQUESTED** [review]

File: packages/bci-whispercpp/index.js (hunk 3 above)
Comment (ishanvohra2) [reply packages/bci-whispercpp/index.js:380]: Resolved.

File: packages/bci-whispercpp/lib/stream.js (hunk 2 above)
Comment (ishanvohra2) [reply packages/bci-whispercpp/lib/stream.js:174]: Resolved.

File: packages/bci-whispercpp/index.js (hunk 1 above)
Comment (ishanvohra2) [reply packages/bci-whispercpp/index.js]: Resolved.

File: packages/bci-whispercpp/CHANGELOG.md (hunk 4 above)
Comment (ishanvohra2) [reply packages/bci-whispercpp/CHANGELOG.md]: Resolved

**Review by GustavoA1604 — APPROVED** [review]

**Review by ogad-tether — APPROVED** [review]
